src/Services/RabbitMQService.cs

```diff
@@ -12,35 +12,33 @@ internal sealed class RabbitMQService : IRabbitMQService
         private readonly IRabbitMQModel _rabbitMQModel;
         private readonly IModel _model;
         private readonly string _connectionString;
-        private readonly string _hostName;
         private readonly string _queueName;
-        private readonly string _userName;
-        private readonly string _password;
-        private readonly int _port;
-        private readonly bool _enableSsl;
-        private readonly bool _skipCertificateValidation;
+        private readonly bool _disableCertificateValidation;
         private readonly object _publishBatchLock;
 
         private IBasicPublishBatch _batch;
 
-        public RabbitMQService(string connectionString, string hostName, string userName, string password, int port, bool enableSsl, bool skipCertificateValidation)
+        public RabbitMQService(string connectionString, bool disableCertificateValidation)
         {
             _connectionString = connectionString;
-            _hostName = hostName;
-            _userName = userName;
-            _password = password;
-            _port = port;
-            _enableSsl = enableSsl;
-            _skipCertificateValidation = skipCertificateValidation;
+            _disableCertificateValidation = disableCertificateValidation;
 
-            ConnectionFactory connectionFactory = GetConnectionFactory(_connectionString, _hostName, _userName, _password, _port, _enableSsl, _skipCertificateValidation);
+            ConnectionFactory connectionFactory = new ConnectionFactory
+            {
+                Uri = new Uri(connectionString),
+            };
+
+            if (disableCertificateValidation && connectionFactory.Ssl.Enabled)
+            {
+                connectionFactory.Ssl.AcceptablePolicyErrors |= SslPolicyErrors.RemoteCertificateChainErrors;
+            }
 

```

Comment:
Method `GetConnectionFactory` is not required. The `connectionFactory.Ssl` (line 31 and 33) will be set to non-null value by RabbitMQ client after we set its `Uri` property (line 28).